backend/internal/test/integration/helper/envoytest/envoy.go

```diff
@@ -106,6 +111,55 @@ func (e *EnvoyHandle) MakeSimpleCall() (int, error) {
 	return resp.StatusCode, nil
 }
 
+// EnsureControlPlaneConnectivity polls the Envoy stats endpoint to ensure that Envoy
+// has an active request against the control plane identified by the provided stat prefix.
+// This is useful in ensuring that Envoy has been able to reconnect to the control plane,
+// even after the exponential backoff that happens as Envoy is unable to connect to the
+// control plane.
+func (e *EnvoyHandle) EnsureControlPlaneConnectivity(prefix string) error {
+	client := &http.Client{}
+
+	r, err := http.NewRequest("GET", "http://envoy:9901/stats", nil)
+	if err != nil {
+		return err
+	}
+
+	timeout := time.NewTimer(20 * time.Second)
+	ticker := time.NewTicker(500 * time.Millisecond)
+
+	for {
+		select {
+		case <-timeout.C:
+			return errors.New("timed out waiting for control plane connectivity")
+		case <-ticker.C:
+			// TODO(snowp): Have this parse out a generic map of stats values to make it easier to query
+			// arbitrary stats.
+			// We intentionally ignore errors here, as the proxy might be periodically unavailable but we
+			// don't care as long as it recovers within the timeout.
+			resp, err := client.Do(r)
+			if err != nil {
+				continue
+			}
+			defer resp.Body.Close()
+
+			allStatsString, _ := ioutil.ReadAll(resp.Body)
```

Comment:
won't close until the entire function returns, you need to explicitly close during each iteration of the loop
```suggestion
			allStatsString, _ := ioutil.ReadAll(resp.Body)
			resp.Body.Close()
```